torch/_dynamo/debug_utils.py

```diff
@@ -962,33 +958,43 @@ def dynamo_accuracy_minifier_backend(gm, example_inputs, compiler_name):
 
     from torch._dynamo.optimizations.backends import BACKENDS
 
+    # (1) Get the backend
     if compiler_name == "inductor":
         from torch._inductor.compile_fx import compile_fx
 
         compiler_fn = compile_fx
     else:
+        assert (
+            compiler_name in BACKENDS
+        ), f"Unknown compiler name {compiler_name} provided."
         compiler_fn = BACKENDS[compiler_name]
 
-    # Set the eval mode to remove randomness.
-    gm.eval()
+    # (2) Compile the graph
+    compiled_gm = compiler_fn(copy.deepcopy(gm), clone_inputs(example_inputs))
 
-    # Check Accuracy
-    if backend_accuracy_fails(gm, example_inputs, compiler_fn):
-        log.warning("Accuracy failed for the TorchDyanmo produced graph")
-        dump_state_fn = functools.partial(
-            dump_backend_state, compiler_name=compiler_name, check_accuracy=True
-        )
-        fails_fn = functools.partial(
-            backend_accuracy_fails,
-            compiler_fn=compiler_fn,
-        )
-        dump_state_fn(fx.GraphModule(gm, copy.deepcopy(gm.graph)), example_inputs)
-        minifier(
-            gm,
-            example_inputs,
-            module_fails=fails_fn,
-            dump_state=dump_state_fn,
-        )
-    else:
-        log.error("Input graph does not fail accuracy testing")
-    return gm
+    def backend_accuracy_eval_fwd(*args):
+        # Set the eval mode to remove randomness.
+        gm.eval()
+
+        def module_fails_accuracy(gm, args):
+            return not same_two_models(gm, compiled_gm, args)
+
+        # (4) Check Accuracy
+        if not same_two_models(gm, compiled_gm, args):
+            log.warning("Accuracy failed for the TorchDyanmo produced graph")
+            dump_state_fn = functools.partial(
+                dump_backend_state, compiler_name=compiler_name, check_accuracy=True
+            )
+
+            dump_state_fn(fx.GraphModule(gm, copy.deepcopy(gm.graph)), args)
+            minifier(
+                gm,
+                example_inputs,
+                module_fails=module_fails_accuracy,
```

Comment:
This looks questionable. The original fails_fn recompiles the passed in gm. However, your fails_fn here simply reuses the original compiled_gm as is. This seems wrong: I'd expect the minifier to be modifying the gm passed in and expecting it to recompile. Without recompiling it I find it hard to see how the minifier could do the correct thing here.

Reply:
Good point, I bet it minifies but not the same way, as the backend is taken out. Let me revisit this.